components/brave_sync/controller_impl.cc

```diff
@@ -0,0 +1,1286 @@
+#include "brave/components/brave_sync/value_debug.h"
+
+#include "chrome/browser/profiles/profile_manager.h"
+#include "chrome/browser/ui/browser_list.h"
+#include "chrome/browser/ui/browser.h"
+#include "content/public/browser/browser_thread.h"
+
+namespace brave_sync {
+
+ControllerImpl::TempStorage::TempStorage() {
+}
+
+ControllerImpl::TempStorage::~TempStorage() {
+}
+
+ControllerImpl::ControllerImpl() :
+  sync_ui_(nullptr),
+  //sync_js_layer_(nullptr),
+  sync_client_(nullptr),
+  sync_initialized_(false),
+  timer_(std::make_unique<base::RepeatingTimer>()) {
+  LOG(ERROR) << "TAGAB brave_sync::ControllerImpl::ControllerImpl CTOR";
+
+  DETACH_FROM_SEQUENCE(sequence_checker_);
+
+  sync_prefs_.reset(new brave_sync::prefs::Prefs(nullptr)); //this is wrong. TODO, AB: pass the pointer
+
+  std::unique_ptr<brave_sync::Settings> settings_ = std::make_unique<brave_sync::Settings>();
+
+  std::unique_ptr<brave_sync::Settings> settingsTest = sync_prefs_->GetBraveSyncSettings();
+  LOG(ERROR) << "TAGAB brave_sync::ControllerImpl::ControllerImpl settingsTest";
+  LOG(ERROR) << "TAGAB brave_sync::ControllerImpl::ControllerImpl settingsTest->this_device_name_=" << settingsTest->this_device_name_;
+  LOG(ERROR) << "TAGAB brave_sync::ControllerImpl::ControllerImpl settingsTest->sync_this_device_=" << settingsTest->sync_this_device_;
+
+  LOG(ERROR) << "TAGAB brave_sync::ControllerImpl::ControllerImpl sync_prefs_->GetSeed()=<" << sync_prefs_->GetSeed() <<">";
+  LOG(ERROR) << "TAGAB brave_sync::ControllerImpl::ControllerImpl sync_prefs_->GetThisDeviceName()=<" << sync_prefs_->GetThisDeviceName() <<">";
+
+  // volatile bool b_DBG_clear_on_start = true;
+  // if (b_DBG_clear_on_start) {
+  //   LOG(ERROR) << "TAGAB brave_sync::ControllerImpl::ControllerImpl clearing prefs";
+  //   sync_prefs_->Clear();
+  // }
+
+  // volatile bool b_DBG_setname_on_start = true;
+  // if (b_DBG_setname_on_start) {
+  //   LOG(ERROR) << "TAGAB brave_sync::ControllerImpl::ControllerImpl set name to cf56";
+  //   sync_prefs_->SetDeviceName("cf56");
+  // LOG(ERROR) << "TAGAB brave_sync::ControllerImpl::ControllerImpl sync_prefs_->GetThisDeviceName()=" << sync_prefs_->GetThisDeviceName();
+  //   DCHECK(false);
+  // }
+
+  task_runner_ = base::CreateSequencedTaskRunnerWithTraits(
+      {base::MayBlock(), base::TaskPriority::BACKGROUND,
+       base::TaskShutdownBehavior::SKIP_ON_SHUTDOWN} );
+
+  sync_obj_map_ = std::make_unique<storage::ObjectMap>();
+
+  bookmarks_ = std::make_unique<brave_sync::Bookmarks>(this);
+
+  if (!sync_prefs_->GetThisDeviceId().empty()) {
+    bookmarks_->SetThisDeviceId(sync_prefs_->GetThisDeviceId());
+  }
+  bookmarks_->SetObjectMap(sync_obj_map_.get());
+
+  BrowserList::GetInstance()->AddObserver(this);
+
+  if (!sync_prefs_->GetSeed().empty() && !sync_prefs_->GetThisDeviceName().empty()) {
+    LOG(ERROR) << "TAGAB brave_sync::ControllerImpl::ControllerImpl sync is configured";
+
+    task_runner_->PostTask(FROM_HERE,
```

Comment:
This is not a thread safe operation. This eventually calls methods that can only be accessed on the UI thread. I'm not sure what you would need a task runner here for at all